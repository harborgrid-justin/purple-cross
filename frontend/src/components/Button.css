/* Button Component Styles */

.btn {
  font-family: inherit;
  font-weight: 500;
  border: none;
  border-radius: var(--radius-md, 0.5rem);
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: var(--spacing-xs, 0.25rem);
  text-decoration: none;
  line-height: 1.5;
}

.btn:focus-visible {
  outline: 2px solid var(--primary-color, #7c3aed);
  outline-offset: 2px;
}

.btn:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

/* Button Sizes */
.btn-sm {
  padding: var(--spacing-xs, 0.25rem) var(--spacing-sm, 0.5rem);
  font-size: 0.875rem;
  min-height: 32px;
}

.btn-md {
  padding: var(--spacing-sm, 0.5rem) var(--spacing-lg, 1.5rem);
  font-size: 1rem;
  min-height: 44px; /* WCAG 2.1 AA touch target */
}

.btn-lg {
  padding: var(--spacing-md, 1rem) var(--spacing-xl, 2rem);
  font-size: 1.125rem;
  min-height: 48px;
}

/* Button Variants */
.btn-primary {
  background-color: var(--primary-color, #7c3aed);
  color: white;
}

.btn-primary:hover:not(:disabled) {
  background-color: var(--primary-dark, #6d28d9);
}

.btn-primary:active:not(:disabled) {
  background-color: #5b21b6;
}

.btn-secondary {
  background-color: var(--bg-primary, #ffffff);
  color: var(--text-primary, #1f2937);
  border: 1px solid var(--border-color, #e5e7eb);
}

.btn-secondary:hover:not(:disabled) {
  background-color: var(--bg-secondary, #f9fafb);
  border-color: var(--text-secondary, #6b7280);
}

.btn-danger {
  background-color: var(--danger-color, #ef4444);
  color: white;
}

.btn-danger:hover:not(:disabled) {
  background-color: #dc2626;
}

.btn-danger:active:not(:disabled) {
  background-color: #b91c1c;
}

.btn-success {
  background-color: var(--secondary-color, #10b981);
  color: white;
}

.btn-success:hover:not(:disabled) {
  background-color: #059669;
}

.btn-success:active:not(:disabled) {
  background-color: #047857;
}

/* Button Modifiers */
.btn-full-width {
  width: 100%;
}

.btn-loading {
  opacity: 0.7;
  cursor: wait;
  pointer-events: none;
}

/* Spinner Animation */
.spinner {
  width: 1em;
  height: 1em;
  border: 2px solid currentColor;
  border-top-color: transparent;
  border-radius: 50%;
  animation: spin 0.6s linear infinite;
}

@keyframes spin {
  to {
    transform: rotate(360deg);
  }
}

/* Reduced Motion Support */
@media (prefers-reduced-motion: reduce) {
  .spinner {
    animation-duration: 1.2s;
  }
}
