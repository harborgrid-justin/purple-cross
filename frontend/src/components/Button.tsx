/**
 * WF-COMP-XXX | Button.tsx - Button
 * Purpose: React component for Button functionality
 * Dependencies: react
 * Last Updated: 2025-10-23 | File Type: .tsx
 */

import { ButtonHTMLAttributes, ReactNode } from 'react';
import clsx from 'clsx';

/**
 * Props for the Button component.
 *
 * @interface ButtonProps
 * @extends {ButtonHTMLAttributes<HTMLButtonElement>}
 *
 * @property {ReactNode} children - The content to display inside the button
 * @property {'primary' | 'secondary' | 'danger' | 'success'} [variant='primary'] - Visual style variant
 * @property {'sm' | 'md' | 'lg'} [size='md'] - Button size
 * @property {boolean} [fullWidth=false] - Whether button should take full width of container
 * @property {boolean} [loading=false] - Loading state - shows spinner and disables button
 *
 * @example
 * // Primary button
 * <Button variant="primary" onClick={handleClick}>
 *   Save Changes
 * </Button>
 *
 * @example
 * // Loading state
 * <Button loading={isSubmitting}>
 *   Submit Form
 * </Button>
 */
interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  variant?: 'primary' | 'secondary' | 'danger' | 'success';
  size?: 'sm' | 'md' | 'lg';
  fullWidth?: boolean;
  loading?: boolean;
}

/**
 * Button component with multiple variants, sizes, and loading state support.
 *
 * @component
 * @param {ButtonProps} props - The component props
 * @returns {JSX.Element} Rendered button element
 *
 * @example
 * // Basic usage
 * <Button onClick={handleSave}>Save</Button>
 *
 * @example
 * // With variant and size
 * <Button variant="danger" size="lg" onClick={handleDelete}>
 *   Delete Account
 * </Button>
 *
 * @example
 * // Full width button
 * <Button fullWidth variant="primary" onClick={handleSubmit}>
 *   Submit Form
 * </Button>
 *
 * @example
 * // Loading state
 * <Button loading={isProcessing} disabled={!isValid}>
 *   Process Payment
 * </Button>
 *
 * @remarks
 * - Extends all native button HTML attributes
 * - Automatically disables button when loading={true}
 * - Uses clsx for conditional className composition
 * - Loading state shows spinner with "Loading..." text
 * - Supports custom className for additional styling
 */
const Button = ({
  children,
  variant = 'primary',
  size = 'md',
  fullWidth = false,
  loading = false,
  disabled,
  className,
  ...props
}: ButtonProps) => {
  return (
    <button
      className={clsx(
        'btn',
        `btn-${variant}`,
        `btn-${size}`,
        {
          'btn-full-width': fullWidth,
          'btn-loading': loading,
        },
        className
      )}
      disabled={disabled || loading}
      {...props}
    >
      {loading ? (
        <>
          <span className="spinner" aria-hidden="true"></span>
          <span>Loading...</span>
        </>
      ) : (
        children
      )}
    </button>
  );
};

export default Button;
