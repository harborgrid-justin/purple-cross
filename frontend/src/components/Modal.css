/* Modal Component Styles */

.modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: var(--z-modal-backdrop, 1040);
  padding: var(--spacing-md, 1rem);
  overflow-y: auto;
}

.modal-content {
  background-color: var(--bg-primary, #ffffff);
  border-radius: var(--radius-lg, 0.75rem);
  box-shadow: var(--shadow-2xl, 0 25px 50px -12px rgba(0, 0, 0, 0.25));
  max-width: 100%;
  max-height: calc(100vh - 2rem);
  overflow: hidden;
  display: flex;
  flex-direction: column;
  position: relative;
  z-index: var(--z-modal, 1050);
}

/* Modal Sizes */
.modal-sm {
  width: 400px;
}

.modal-md {
  width: 600px;
}

.modal-lg {
  width: 800px;
}

.modal-xl {
  width: 1200px;
}

.modal-full {
  width: calc(100vw - 2rem);
  height: calc(100vh - 2rem);
}

/* Modal Header */
.modal-header {
  padding: var(--spacing-lg, 1.5rem);
  border-bottom: 1px solid var(--border-color, #e5e7eb);
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-shrink: 0;
}

.modal-title {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 600;
  color: var(--text-primary, #1f2937);
  line-height: 1.5;
}

.modal-close {
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0.5rem;
  border-radius: var(--radius-sm, 0.25rem);
  color: var(--text-secondary, #6b7280);
  transition: background-color 0.2s, color 0.2s;
  font-size: 1.5rem;
  line-height: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 2.5rem;
  height: 2.5rem;
  margin: -0.5rem -0.5rem -0.5rem 0;
}

.modal-close:hover {
  background-color: var(--bg-secondary, #f9fafb);
  color: var(--text-primary, #1f2937);
}

.modal-close:focus-visible {
  outline: 2px solid var(--primary-color, #7c3aed);
  outline-offset: 2px;
}

/* Modal Body */
.modal-body {
  padding: var(--spacing-lg, 1.5rem);
  overflow-y: auto;
  flex: 1;
}

.modal-body-no-padding {
  padding: 0;
}

/* Modal Footer */
.modal-footer {
  padding: var(--spacing-lg, 1.5rem);
  border-top: 1px solid var(--border-color, #e5e7eb);
  display: flex;
  align-items: center;
  justify-content: flex-end;
  gap: var(--spacing-md, 1rem);
  flex-shrink: 0;
  background-color: var(--bg-secondary, #f9fafb);
}

/* Animations */
@keyframes modalFadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes modalSlideUp {
  from {
    opacity: 0;
    transform: translateY(20px) scale(0.95);
  }
  to {
    opacity: 1;
    transform: translateY(0) scale(1);
  }
}

.modal-overlay {
  animation: modalFadeIn 0.2s ease-out;
}

.modal-content {
  animation: modalSlideUp 0.2s ease-out;
}

/* Responsive */
@media (max-width: 768px) {
  .modal-overlay {
    padding: 0;
    align-items: flex-end;
  }

  .modal-content {
    border-radius: var(--radius-lg, 0.75rem) var(--radius-lg, 0.75rem) 0 0;
    max-height: 90vh;
  }

  .modal-sm,
  .modal-md,
  .modal-lg,
  .modal-xl {
    width: 100%;
  }

  .modal-header,
  .modal-body,
  .modal-footer {
    padding: var(--spacing-md, 1rem);
  }

  .modal-title {
    font-size: 1.125rem;
  }
}

/* Reduced Motion Support */
@media (prefers-reduced-motion: reduce) {
  .modal-overlay,
  .modal-content {
    animation: none;
  }
}

/* Backdrop Click Helper */
.modal-backdrop-click-area {
  position: absolute;
  inset: 0;
  z-index: -1;
}
