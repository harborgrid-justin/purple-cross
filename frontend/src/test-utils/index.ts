/**
 * WF-COMP-XXX | index.ts - Test Utils Exports
 * Purpose: Central export point for all test utilities
 * Dependencies: None
 * Last Updated: 2025-10-24 | File Type: .ts
 */

// Export custom render functions
export * from './render';

// Export test helpers
export * from './helpers';

// Export test fixtures
export * from './fixtures';

// Export MSW mocks
export * from './mocks/handlers';
export { server, mockServer } from './mocks/server';
