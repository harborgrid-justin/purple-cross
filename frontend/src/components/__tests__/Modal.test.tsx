/**
 * WF-COMP-XXX | Modal.test.tsx - Modal Component Tests
 * Purpose: Comprehensive tests for Modal component
 * Dependencies: @testing-library/react, @testing-library/user-event
 * Last Updated: 2025-10-24 | File Type: .tsx
 */

import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
import { renderWithoutRouter, createUser } from '@/test-utils';
import Modal from '../Modal';
import Button from '../Button';

describe('Modal Component', () => {
  let originalBodyOverflow: string;

  beforeEach(() => {
    // Save original body overflow style
    originalBodyOverflow = document.body.style.overflow;
  });

  afterEach(() => {
    // Restore original body overflow style
    document.body.style.overflow = originalBodyOverflow;
  });

  describe('Rendering', () => {
    it('should not render when isOpen is false', () => {
      const { queryByRole } = renderWithoutRouter(
        <Modal isOpen={false} onClose={vi.fn()}>
          <p>Modal content</p>
        </Modal>
      );

      expect(queryByRole('dialog')).not.toBeInTheDocument();
    });

    it('should render when isOpen is true', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Modal content</p>
        </Modal>
      );

      expect(getByRole('dialog')).toBeInTheDocument();
    });

    it('should render children correctly', () => {
      const { getByText } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Modal content</p>
        </Modal>
      );

      expect(getByText('Modal content')).toBeInTheDocument();
    });

    it('should render modal overlay', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const overlay = container.querySelector('.modal-overlay');
      expect(overlay).toBeInTheDocument();
    });

    it('should render modal content', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const content = container.querySelector('.modal-content');
      expect(content).toBeInTheDocument();
    });

    it('should render modal body', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const body = container.querySelector('.modal-body');
      expect(body).toBeInTheDocument();
    });
  });

  describe('Title', () => {
    it('should render title when provided', () => {
      const { getByText } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Modal Title">
          <p>Content</p>
        </Modal>
      );

      expect(getByText('Modal Title')).toBeInTheDocument();
    });

    it('should render title as h2 heading', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Modal Title">
          <p>Content</p>
        </Modal>
      );

      const heading = getByRole('heading', { level: 2 });
      expect(heading).toHaveTextContent('Modal Title');
    });

    it('should have correct title id', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Modal Title">
          <p>Content</p>
        </Modal>
      );

      const title = container.querySelector('#modal-title');
      expect(title).toBeInTheDocument();
      expect(title).toHaveTextContent('Modal Title');
    });

    it('should have correct title class', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Modal Title">
          <p>Content</p>
        </Modal>
      );

      const title = container.querySelector('.modal-title');
      expect(title).toBeInTheDocument();
    });

    it('should not render title when not provided', () => {
      const { queryByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      expect(queryByRole('heading')).not.toBeInTheDocument();
    });

    it('should connect title with dialog via aria-labelledby', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Modal Title">
          <p>Content</p>
        </Modal>
      );

      const dialog = getByRole('dialog');
      expect(dialog).toHaveAttribute('aria-labelledby', 'modal-title');
    });

    it('should not have aria-labelledby when no title', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const dialog = getByRole('dialog');
      expect(dialog).not.toHaveAttribute('aria-labelledby');
    });
  });

  describe('Close Button', () => {
    it('should render close button', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      expect(getByRole('button', { name: /close modal/i })).toBeInTheDocument();
    });

    it('should have correct aria-label on close button', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const closeButton = getByRole('button', { name: /close modal/i });
      expect(closeButton).toHaveAttribute('aria-label', 'Close modal');
    });

    it('should have correct type on close button', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const closeButton = getByRole('button', { name: /close modal/i });
      expect(closeButton).toHaveAttribute('type', 'button');
    });

    it('should display × symbol in close button', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const closeButton = getByRole('button', { name: /close modal/i });
      expect(closeButton).toHaveTextContent('×');
    });

    it('should have modal-close class', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const closeButton = getByRole('button', { name: /close modal/i });
      expect(closeButton).toHaveClass('modal-close');
    });

    it('should call onClose when close button is clicked', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={handleClose}>
          <p>Content</p>
        </Modal>
      );

      const closeButton = getByRole('button', { name: /close modal/i });
      await user.click(closeButton);

      expect(handleClose).toHaveBeenCalledTimes(1);
    });

    it('should support keyboard interaction (Enter)', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={handleClose}>
          <p>Content</p>
        </Modal>
      );

      const closeButton = getByRole('button', { name: /close modal/i });
      closeButton.focus();
      await user.keyboard('{Enter}');

      expect(handleClose).toHaveBeenCalledTimes(1);
    });
  });

  describe('Footer', () => {
    it('should render footer when provided', () => {
      const footer = (
        <>
          <Button>Cancel</Button>
          <Button variant="primary">Save</Button>
        </>
      );
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} footer={footer}>
          <p>Content</p>
        </Modal>
      );

      const modalFooter = container.querySelector('.modal-footer');
      expect(modalFooter).toBeInTheDocument();
    });

    it('should not render footer when not provided', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const modalFooter = container.querySelector('.modal-footer');
      expect(modalFooter).not.toBeInTheDocument();
    });

    it('should render footer buttons', () => {
      const footer = (
        <>
          <Button>Cancel</Button>
          <Button variant="primary">Save</Button>
        </>
      );
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} footer={footer}>
          <p>Content</p>
        </Modal>
      );

      expect(getByRole('button', { name: /cancel/i })).toBeInTheDocument();
      expect(getByRole('button', { name: /save/i })).toBeInTheDocument();
    });
  });

  describe('Sizes', () => {
    it('should render with small size', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} size="sm">
          <p>Content</p>
        </Modal>
      );

      const content = container.querySelector('.modal-content');
      expect(content).toHaveClass('modal-sm');
    });

    it('should render with medium size (default)', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} size="md">
          <p>Content</p>
        </Modal>
      );

      const content = container.querySelector('.modal-content');
      expect(content).toHaveClass('modal-md');
    });

    it('should render with large size', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} size="lg">
          <p>Content</p>
        </Modal>
      );

      const content = container.querySelector('.modal-content');
      expect(content).toHaveClass('modal-lg');
    });

    it('should render with extra large size', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} size="xl">
          <p>Content</p>
        </Modal>
      );

      const content = container.querySelector('.modal-content');
      expect(content).toHaveClass('modal-xl');
    });

    it('should default to medium size', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const content = container.querySelector('.modal-content');
      expect(content).toHaveClass('modal-md');
    });
  });

  describe('Overlay Click Behavior', () => {
    it('should close on overlay click by default', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={handleClose}>
          <p>Content</p>
        </Modal>
      );

      const overlay = container.querySelector('.modal-overlay') as HTMLElement;
      await user.click(overlay);

      expect(handleClose).toHaveBeenCalledTimes(1);
    });

    it('should close on overlay click when closeOnOverlayClick is true', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={handleClose} closeOnOverlayClick>
          <p>Content</p>
        </Modal>
      );

      const overlay = container.querySelector('.modal-overlay') as HTMLElement;
      await user.click(overlay);

      expect(handleClose).toHaveBeenCalledTimes(1);
    });

    it('should not close on overlay click when closeOnOverlayClick is false', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={handleClose} closeOnOverlayClick={false}>
          <p>Content</p>
        </Modal>
      );

      const overlay = container.querySelector('.modal-overlay') as HTMLElement;
      await user.click(overlay);

      expect(handleClose).not.toHaveBeenCalled();
    });

    it('should not close when clicking modal content', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={handleClose}>
          <p>Content</p>
        </Modal>
      );

      const content = container.querySelector('.modal-content') as HTMLElement;
      await user.click(content);

      expect(handleClose).not.toHaveBeenCalled();
    });

    it('should not close when clicking modal body', async () => {
      const handleClose = vi.fn();
      const user = createUser();
      const { getByText } = renderWithoutRouter(
        <Modal isOpen onClose={handleClose}>
          <p>Click me</p>
        </Modal>
      );

      await user.click(getByText('Click me'));

      expect(handleClose).not.toHaveBeenCalled();
    });
  });

  describe('Body Scroll Lock', () => {
    it('should lock body scroll when modal opens', () => {
      renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      expect(document.body.style.overflow).toBe('hidden');
    });

    it('should unlock body scroll when modal closes', () => {
      const { rerender } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      expect(document.body.style.overflow).toBe('hidden');

      rerender(
        <Modal isOpen={false} onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      expect(document.body.style.overflow).toBe('unset');
    });

    it('should restore body scroll on unmount', () => {
      const { unmount } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      expect(document.body.style.overflow).toBe('hidden');

      unmount();

      expect(document.body.style.overflow).toBe('unset');
    });
  });

  describe('Modal Structure', () => {
    it('should render modal header', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Title">
          <p>Content</p>
        </Modal>
      );

      const header = container.querySelector('.modal-header');
      expect(header).toBeInTheDocument();
    });

    it('should contain title and close button in header', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Title">
          <p>Content</p>
        </Modal>
      );

      const header = container.querySelector('.modal-header');
      const title = container.querySelector('.modal-title');
      const closeButton = container.querySelector('.modal-close');

      expect(header).toContainElement(title);
      expect(header).toContainElement(closeButton);
    });

    it('should render all parts in correct order', () => {
      const footer = <Button>Close</Button>;
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Title" footer={footer}>
          <p>Content</p>
        </Modal>
      );

      const content = container.querySelector('.modal-content');
      const children = Array.from(content?.children || []);

      expect(children[0]).toHaveClass('modal-header');
      expect(children[1]).toHaveClass('modal-body');
      expect(children[2]).toHaveClass('modal-footer');
    });
  });

  describe('Accessibility', () => {
    it('should have role="dialog"', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      expect(getByRole('dialog')).toBeInTheDocument();
    });

    it('should have aria-modal="true"', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const dialog = getByRole('dialog');
      expect(dialog).toHaveAttribute('aria-modal', 'true');
    });

    it('should connect title with dialog via aria-labelledby', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Modal Title">
          <p>Content</p>
        </Modal>
      );

      const dialog = getByRole('dialog');
      expect(dialog).toHaveAttribute('aria-labelledby', 'modal-title');
    });

    it('should have accessible close button', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      const closeButton = getByRole('button', { name: /close modal/i });
      expect(closeButton).toHaveAccessibleName('Close modal');
    });

    it('should be keyboard navigable', async () => {
      const user = createUser();
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <Button>Action</Button>
        </Modal>
      );

      await user.tab();

      const button = getByRole('button', { name: /action/i });
      expect(button).toHaveFocus();
    });
  });

  describe('Content Rendering', () => {
    it('should render simple text content', () => {
      const { getByText } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Simple text</p>
        </Modal>
      );

      expect(getByText('Simple text')).toBeInTheDocument();
    });

    it('should render complex nested content', () => {
      const { getByText, getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Complex Modal">
          <div>
            <h3>Section</h3>
            <p>Paragraph</p>
            <Button>Action</Button>
          </div>
        </Modal>
      );

      expect(getByText('Section')).toBeInTheDocument();
      expect(getByText('Paragraph')).toBeInTheDocument();
      expect(getByRole('button', { name: /action/i })).toBeInTheDocument();
    });

    it('should render forms', () => {
      const { getByLabelText, getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Form Modal">
          <form>
            <label htmlFor="name">Name</label>
            <input id="name" type="text" />
            <Button type="submit">Submit</Button>
          </form>
        </Modal>
      );

      expect(getByLabelText('Name')).toBeInTheDocument();
      expect(getByRole('button', { name: /submit/i })).toBeInTheDocument();
    });
  });

  describe('Edge Cases', () => {
    it('should handle empty children', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} />
      );

      expect(getByRole('dialog')).toBeInTheDocument();
    });

    it('should handle very long titles', () => {
      const longTitle =
        'This is a very long modal title that might need to wrap across multiple lines';
      const { getByText } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title={longTitle}>
          <p>Content</p>
        </Modal>
      );

      expect(getByText(longTitle)).toBeInTheDocument();
    });

    it('should handle null children gracefully', () => {
      const { getByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()} title="Title">
          {null}
        </Modal>
      );

      expect(getByRole('dialog')).toBeInTheDocument();
    });

    it('should not render anything when isOpen changes from true to false', () => {
      const { rerender, queryByRole } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      expect(queryByRole('dialog')).toBeInTheDocument();

      rerender(
        <Modal isOpen={false} onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );

      expect(queryByRole('dialog')).not.toBeInTheDocument();
    });
  });

  describe('Snapshot Tests', () => {
    it('should match snapshot for simple modal', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen onClose={vi.fn()}>
          <p>Simple content</p>
        </Modal>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for modal with all features', () => {
      const footer = <Button>Close</Button>;
      const { container } = renderWithoutRouter(
        <Modal
          isOpen
          onClose={vi.fn()}
          title="Full Featured Modal"
          footer={footer}
          size="lg"
        >
          <p>Modal content</p>
        </Modal>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot when closed', () => {
      const { container } = renderWithoutRouter(
        <Modal isOpen={false} onClose={vi.fn()}>
          <p>Content</p>
        </Modal>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
