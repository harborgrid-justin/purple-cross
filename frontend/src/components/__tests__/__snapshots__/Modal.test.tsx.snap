// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal Component > Snapshot Tests > should match snapshot for modal with all features 1`] = `
<div
  aria-labelledby="modal-title"
  aria-modal="true"
  class="modal-overlay"
  role="dialog"
>
  <div
    class="modal-content modal-lg"
  >
    <div
      class="modal-header"
    >
      <h2
        class="modal-title"
        id="modal-title"
      >
        Full Featured Modal
      </h2>
      <button
        aria-label="Close modal"
        class="modal-close"
        type="button"
      >
        ×
      </button>
    </div>
    <div
      class="modal-body"
    >
      <p>
        Modal content
      </p>
    </div>
    <div
      class="modal-footer"
    >
      <button
        class="btn btn-primary btn-md"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`Modal Component > Snapshot Tests > should match snapshot for simple modal 1`] = `
<div
  aria-modal="true"
  class="modal-overlay"
  role="dialog"
>
  <div
    class="modal-content modal-md"
  >
    <div
      class="modal-header"
    >
      <button
        aria-label="Close modal"
        class="modal-close"
        type="button"
      >
        ×
      </button>
    </div>
    <div
      class="modal-body"
    >
      <p>
        Simple content
      </p>
    </div>
  </div>
</div>
`;

exports[`Modal Component > Snapshot Tests > should match snapshot when closed 1`] = `null`;
